pywbemtools/pywbemcli/pywbemcli.py

```diff
@@ -255,14 +255,87 @@ def _validate_no_server_option_conficts(server, connection_name, mock_server):
+
+            # Test if the default connection name is actually
+            # in the repo.  If not reset the default connection
+            # name to None and continue.
+            # This should never occur unless connection file
+            # is corrupted.
+            if connection_name and connection_name not in \
+                    connections_repo:
+                connections_repo.default_connection_name = None
+                raise click.ClickException(
+                    'Default connection name: "{}" not found in '
+                    'connections file "{}"; default connection '
+                    'name cleared.'.
+                    format(connection_name,
+                           connections_repo.connections_file))
+            if verbose:
+                click.echo('Current connection: "{}"'
+                           .format(connection_name))
+            return connection_name
+
+        # returns no if there is no file
+        return None
+
+    # Account for other exception from file access
+    except ConnectionsFileError as cfc:
+        click.echo("{}, {}". format(cfc.__class__.__name__, cfc),
+                   err=True)
+        raise click.Abort()
+
+
+def _validate_server_only_options(pywbem_server, user, password, timeout,
+                                  verify, certfile, keyfile, ca_certs):
+    """
+    Validate that the options defined as parameters for this function
+    are not set if the server type is mock_server
+    """
+    if pywbem_server.mock_server:
+        opts = [user, password, timeout, verify is not None, certfile,
+                keyfile, ca_certs]
+        if any(opts):
+            opt_names = ["--user", "--password", "--timeout",
+                         "--verify", "--certfile", "--keyfile",
+                         "--ca_certs"]
+            errors = [opt_names[c] for c, i in enumerate(opts) if i]
+            raise click.ClickException(
+                "General option(s): ({0}) not allowed with --mock-server "
+                "general option'".
+                format(", ".join(errors)))
+
+
+def _create_server_instance(server, connection_name,
+                            resolved_default_namespace,
+                            user, password,
+                            resolved_timeout,
+                            resolved_use_pull,
+                            resolved_pull_max_cnt,
                             resolved_verify,
-                            certfile, keyfile, resolved_ca_certs,
-                            resolved_mock_server, connections_repo,
-                            conditional_options,
-                            verbose):
+                            certfile, keyfile, ca_certs,
+                            resolved_mock_server,
+                            connections_repo,
+                            verbose,
+                            default_namespace, use_pull, pull_max_cnt,
+                            timeout, verify):
```

Comment:
This layout is a pain when debugging. I suggest to split already the first parameter into a new line indented by 8, and then fill the lines to the end to minimize the number of lines.

Reply:
Agree completely but it is very good for establishing that all parameters are in place.  Changed to your suggestion now that we know all are in place.